File: 24-Java/src/Ricerca.java
public class Ricerca extends Progetto{
    private String argomento;
    private int numPartner;
    
    public Ricerca(int codice, String tipo, String titolo, String nomeCognome, String organizzazione, double milioni,
            String argomento, int numPartner) {
        super(codice, tipo, titolo, nomeCognome, organizzazione, milioni);
        this.argomento = argomento;
        this.numPartner = numPartner;
    }

    public String toString(){
        return super.toString() +"\t"+ argomento +"\t"+ numPartner +"\t-\t"+ super.milioni*1000;
    }
}

File: 24-Python/Ricerca.py
from Progetto import Progetto

class Ricerca(Progetto):
    def __init__(self, codice, tipo, titolo, nomeCognome, coordinatore, milioni, argomento, numPartner) -> None:
        super().__init__(codice, tipo, titolo, nomeCognome, coordinatore, milioni)
        self._argomento = argomento
        self._numPartner = numPartner

    def __str__(self) -> str:
        return super().__str__() +"\t"+ self._argomento +"\t"+ str(self._numPartner) +"\t-\t"+ str(self._milioni * 1000)